pandas/core/strings.py

```diff
@@ -45,6 +45,15 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
 
     Examples
     --------
+    When ``na_rep`` is `None` (default behavior), NaN value(s)
+    in the Series propagate and return value will be NaN.
+
+    >>> Series(['a','b',np.nan,'c']).str.cat(sep=' ')
```

Comment:
actually you bring up a good point. doing ``.fillna(na_rep)``, THEN processing might just work and do the right thing.

yes if ``others`` is then inconsistent you need to raise.

pls add as many test cases as you think are needed to represent all behavior.